packages/custodyKeyring/src/custodianTypes/qredo/QredoCustodyKeyring.ts

```diff
@@ -59,4 +60,8 @@ export class QredoCustodyKeyring extends CustodyKeyring {
   getStatusMap(): ITransactionStatusMap {
     return QredoStatusMap;
   }
+
+  protected getCustodianFromEnvName(envName: string): ICustodianEnvironment {
+    return this.getCustodians().find(c => /qredo/iu.test(c.envName));
```

Comment:
This looks correct, I'm just not sure if it will be called instead of the main one in CustodyKeyring. So in the build I've added to the `getCustodianFromEnvName` from CustodyKeyring the following:

```
    getCustodianFromEnvName(envName) {
        let newEnvName = envName;
        if (!newEnvName) {
            if (this.type === 'Custody - Qredo') {
                newEnvName = 'qredo'
            } else {
                return {};
            }
        }
        
       const custodian = this.getCustodians().find(c => c.envName === newEnvName);

        ....
```

And next time, I'll write it cleaner

Reply:
thats a good point, I think it depends on the visibilty of the method? but we can figure that out when its calmer

Reply:
*property